src/v/storage/segment_appender.cc

```diff
@@ -336,15 +336,21 @@ ss::future<> segment_appender::do_next_adaptive_fallocation() {
                  return _out.allocate(_fallocation_offset, step)
                    .then([this, step] { _fallocation_offset += step; });
              })
-      .handle_exception([this](std::exception_ptr e) {
-          vassert(
-            false,
+      .handle_exception_type([this](std::system_error e) {
+          vlog(stlog.error,
             "We failed to fallocate file. This usually means we have ran out "
             "of disk space. Please check your data partition and ensure you "
             "have enough space. Error: {} - {}",
-            e,
-            *this);
+            e, *this);
+          ss::throw_with_backtrace<std::system_error>(e);
       });
+      // Would like to catch specific type above then the rest here.
```

Comment:
I can't chain another `.handle_exception()` to this with a catch-all `std::exception_ptr`, obviously, since I rethrow in the previous link, and it would catch it. I guess I want `.handle_exception_type(anything but std::system_error)`.. Any tips?